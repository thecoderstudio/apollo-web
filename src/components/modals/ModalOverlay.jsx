import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const propTypes = {
  children: PropTypes.node.isRequired,
};

const Overlay = styled.div`
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: ${props => props.theme.overlay};
`;

export default function ModalOverlay(props) {
  return (
    <Overlay>
      {props.children}
    </Overlay>
  );
}

ModalOverlay.propTypes = propTypes;

