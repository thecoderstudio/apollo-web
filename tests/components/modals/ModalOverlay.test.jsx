import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ModalOverlay from '../../../src/components/modals/ModalOverlay';


function getComponent(spy=jest.fn()) {
  return renderer.create(
    <ModalOverlay closeModalFunction={spy}>
      <div />
    </ModalOverlay>
  );
}

describe('ModalOverlay', () => {
  it('renders correctly', () => {
    expect(getComponent().toJSON()).toMatchSnapshot();
  });

  it('calls close function', () => {
    const spy = jest.fn();
    const wrapper = mount(
      <ModalOverlay closeModalFunction={spy}>
        <div />
      </ModalOverlay>
    );
    wrapper.find('div').first().simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('does not call close function', () => {
    const spy = jest.fn();
    const wrapper = mount(
      <ModalOverlay closeModalFunction={spy}>
        <div id='test' />
      </ModalOverlay>
    );
    wrapper.find('#test').simulate('click');
    expect(spy).toHaveBeenCalledTimes(0);
  });
});
