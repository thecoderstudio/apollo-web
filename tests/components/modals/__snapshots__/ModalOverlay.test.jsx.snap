// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalOverlay renders correctly 1`] = `
.c0 {
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
}

<div
  className="c0"
  onClick={[Function]}
>
  <div />
</div>
`;
